src/nlsolve/anderson.jl

```diff
@@ -0,0 +1,67 @@
+function (S::NLAnderson{false,<:NLSolverCache})(integrator)
+  nlcache = S.cache
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack z,tmp,W,κ,tol,c,γ,max_iter,min_iter = nlcache
+  if typeof(integrator.f) <: SplitFunction
+    f = integrator.f.f1
+  else
+    f = integrator.f
+  end
+  # precalculations
+  κtol = κ*tol
+
+  zs = zeros(S.n+1)
+  gs = zeros(S.n+1)
```

Comment:
Shouldn't these definitions of `zs` and `gs` cause problems if `f` returns general number types and/or vectors? I realized that the current implementation in https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/nlsolve/functional.jl#L25 only considers the scalar case - is there a reason for not considering vector valued derivatives? I guess, the implementation there should be fixed (i.e. generalized) as well.

Reply:
Basically the generalization should only require adding dots to all possibly vector valued computations.

Reply:
@devmotion The dots are there in the NLFunctional{true} implementation, (see here: https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/nlsolve/newton.jl#L87 ). I have currently done the same for the sake of uniformity. Implementation for `iip=true` is to be done in the coming commits, but we for sure can work to generalize it.

Reply:
All out-of-place solvers don't have broadcasts, so I also didn't add dots in functional iteration.

https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/perform_step/fixed_timestep_perform_step.jl#L181-L182

Reply:
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/582 merged so this PR should take it into consideration.